ludwig/api.py

```diff
@@ -519,25 +505,32 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                 random_seed=random_seed,
             ) as trainer:
                 # auto tune batch size
-                if self.config[TRAINER].get(BATCH_SIZE, None) == AUTO or self.config[TRAINER][EVAL_BATCH_SIZE] == AUTO:
+                if (
+                    self.config_obj.trainer.to_dict().get(BATCH_SIZE, None) == AUTO
+                    or self.config_obj.trainer.to_dict().get(EVAL_BATCH_SIZE, None) == AUTO
+                ):
                     # TODO (ASN): add support for substitute_with_max parameter
                     # TODO(travis): detect train and eval batch sizes separately (enable / disable gradients)
-                    tuned_batch_size = trainer.tune_batch_size(self.config, training_set, random_seed=random_seed)
+                    tuned_batch_size = trainer.tune_batch_size(
+                        self.config_obj.to_dict(), training_set, random_seed=random_seed
+                    )
 
                     # TODO(travis): pass these in as args to trainer when we call train,
                     #  to avoid setting state on possibly remote trainer
-                    if self.config[TRAINER][BATCH_SIZE] == AUTO:
-                        self.config[TRAINER][BATCH_SIZE] = tuned_batch_size
+                    if self.config_obj.trainer.to_dict().get(BATCH_SIZE, None) == AUTO:
```

Comment:
Reason I had it like this is because GBM trainer does not have a batch size param so if we tried to index with dot notation, I thought it would have a chance of throwing an error. Upon looking closer, I believe you are right since it won't even enter the if clause unless it's an ECD model since you cannot set `eval_batch_size` to `auto` in a gbm model so I will make the change. Good eye.